keps/sig-node/3085-pod-conditions-for-starting-completition-of-sandbox-creation/README.md

```diff
@@ -0,0 +1,1430 @@
+  - [ ] e2e Tests for all Beta API Operations (endpoints)
+  - [ ] (R) Ensure GA e2e tests for meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md)
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [X] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md)
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+<!--
+This section is incredibly important for producing high-quality, user-focused
+documentation such as release notes or a development roadmap. It should be
+possible to collect this information before implementation begins, in order to
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself. KEP editors and SIG Docs
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+Pod sandbox creation is a critical phase of a pod's lifecycle that the kubelet
+orchestrates across multiple components: in-tree volume plugins (ConfigMap,
+Secret, EmptyDir, etc), CSI plugins, container runtime (which in turn invokes
+CNI plugins) and associated runtime handler and other components. This KEP
+proposes a `SandboxReady` condition in pod status to indicate successful
+completion of pod sandbox creation by Kubelet. The `SandboxReady` condition will
+mark an important milestone in the pod's lifecycle similar to `ContainersReady`
+and the overall `Ready` conditions in pod status today.
+
+## Motivation
```

Comment:
Perhaps, another usecase is that SandboxReady can indicate if the sandbox cleanup is done. I refer to the CRI function `StopPodSandbox`. If `StopPodSandbox` keeps failing, kubelet can keep SandboxReady as true.

Do you think this makes sense?

Reply:
I may confuse `SandboxReady:false` with sandbox teardown which you mentioned as a non-goal. 

But I think we need to specify when `SandboxReady` is set to false during pod termination.

Reply:
Clarified the non-goal a bit by stating: `Specify additional conditions (beyond SandboxReady with status set to false) to indicate sandbox teardown`. Let me know if this is better.

Added details in `SandboxReady condition details` section covering how the invocation of `generateAPIPodStatus` after killPod() will result in `SandboxReady` becoming `false`.